pkg/config/DeploymentConfigurationService.go

```diff
@@ -0,0 +1,67 @@
+package config
+
+import (
+	"github.com/devtron-labs/devtron/internal/sql/repository/chartConfig"
+	"github.com/devtron-labs/devtron/pkg/pipeline"
+	"go.uber.org/zap"
+)
+
+type DeploymentConfigurationService interface {
+	ConfigAutoComplete(appId int, envId int) (*ConfigDataResponse, error)
+}
+
+type DeploymentConfigurationServiceImpl struct {
+	logger           *zap.SugaredLogger
+	configMapService pipeline.ConfigMapService
+}
+
+func NewDeploymentConfigurationServiceImpl(logger *zap.SugaredLogger,
+	configMapService pipeline.ConfigMapService,
+) (*DeploymentConfigurationServiceImpl, error) {
+	deploymentConfigurationService := &DeploymentConfigurationServiceImpl{
+		logger:           logger,
+		configMapService: configMapService,
+	}
+
+	return deploymentConfigurationService, nil
+}
+
+func (impl *DeploymentConfigurationServiceImpl) ConfigAutoComplete(appId int, envId int) (*ConfigDataResponse, error) {
+	var configDataResponse *ConfigDataResponse
+	cMCSNamesAppLevel, cMCSNamesEnvLevel, err := impl.configMapService.FetchCmCsNamesAppAndEnvLevel(appId, envId)
+	if err != nil {
+		return nil, err
+	}
+	configDataResponse = setConfigDataResponse(cMCSNamesAppLevel, configDataResponse)
+	configDataResponse = setConfigDataResponse(cMCSNamesEnvLevel, configDataResponse)
```

Comment:
this would append both app level and app-env level config maps and secrets and we will get duplicates